### PR TITLE
ServiceTicket Open (cim#10)

### PR DESCRIPTION
**Pull request by Mark90**
### `src/`
* Change UserInputForm to allow overriding the text and color of previous/next buttons, and to show a dialog for the previous button
* Make the order of custompages consistent

### `src/custom-surf/`
* Cim serviceticket Open : Add ServiceTicketOpen page and OpenForm component
* Cim serviceticket Create: Change `title` to `title_nl`
* Cim serviceticket Detail: fix Opening button requiredState
* Cim servicetickets: adjust column widths

### CODE REVIEW

File: src/custom-surf/pages/ServiceTicketOpen.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            cim_ticket_id: id,
+            title_nl: userInputs.title_nl,
+            description_nl: userInputs.description_nl,
+            title_en: userInputs.title_en,
+            description_en: userInputs.description_en,
+            mail_subject: userInputs.mail_subject,
+        };
+        // TODO:
+        //   - ask for confirmation here?
+        //   - or suggestion by Rene, redirect to other page for displaying the emails?
```
Comment (Mark90) [inline src/custom-surf/pages/ServiceTicketOpen.tsx:45]: todo; check if we can have a simple popup here that describes what's going to happen

File: src/custom-surf/components/cim/OpenForm.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+                redirect("/tickets");
+            });
+        },
+        [redirect, intl, customApiClient, formKey, handleSubmit]
+    );
+
+    const cancel = useCallback(() => {
+        // TODO ask user to confirm leaving the page
+        handleCancel();
+    }, [redirect, intl, customApiClient, formKey, handleCancel]);
```
Comment (tjeerddie) [inline src/custom-surf/components/cim/OpenForm.tsx:63]: are the `redirect, intl, customApiClient, formKey,` necessary?

File: src/custom-surf/components/cim/OpenForm.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+interface IProps {
+    formKey: string;
+    ticket_id: string;
+    handleSubmit: (userInputs: any) => void;
+    handleCancel: () => void;
+}
+
+export default function OpenForm(props: IProps) {
+    const intl = useIntl();
+    const { formKey, ticket_id, handleSubmit, handleCancel } = props;
```
Comment (tjeerddie) [inline src/custom-surf/components/cim/OpenForm.tsx:33]: can be done directly within the function
`export default function OpenForm({ formKey, ticket_id, handleSubmit, handleCancel }: IProps) {`

File: src/custom-surf/pages/CreateServiceTicket.tsx [hunk 4]
```diff
@@ -28,7 +28,7 @@ export default function CreateServiceTicket() {
         const ticket: CreateServiceTicketPayload = {
             ims_pw_id: userInputs.ims_ticket.name,
             jira_ticket_id: userInputs.jira_ticket.ticket_id,
-            title: userInputs.jira_ticket.summary,
+            title_nl: userInputs.jira_ticket.summary,
```
Comment (tjeerddie) [inline src/custom-surf/pages/CreateServiceTicket.tsx:31]: the `title_en` not necessary?

File: src/custom-surf/components/cim/OpenForm.tsx (hunk 2 above)
Comment (Mark90) [reply src/custom-surf/components/cim/OpenForm.tsx:63]: Nope, removing them

File: src/custom-surf/pages/ServiceTickets.tsx [hunk 5]
```diff
@@ -262,16 +262,16 @@ class ServiceTickets extends React.PureComponent<IProps, IState> {
...
-                                        className="subscription"
+                                        className="jira_ticket_id"
                                         onClick={() => this.context.redirect(`/tickets/${ticket._id}`)}
                                     >
                                         {ticket.jira_ticket_id}
                                     </td>
                                     <td
                                         data-label={intl.formatMessage({ id: "tickets.table.title" })}
-                                        className="customer"
+                                        className="title"
```
[the remark this answers is not in the document]
Comment (Mark90) [reply src/custom-surf/pages/ServiceTickets.tsx:272]: Updated all th styles now

File: src/custom-surf/pages/ServiceTicketOpen.tsx (hunk 1 above)
Comment (Mark90) [reply src/custom-surf/pages/ServiceTicketOpen.tsx]: Submitting the underlying form from a popup modal proved too difficult for me. Workaround is that the final form submit button has a different color and explicit text `Going to send mail to <number> institutes`.

**Review by acidjunk — APPROVED** [review]
LGTM

**Pull request merged**